hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/upgrade/EightToSevenDowngradeHandler.java

```diff
@@ -92,4 +115,52 @@ private static void downgradePartitionFields(HoodieWriteConfig config, HoodieEng
       tablePropsToAdd.put(HoodieTableConfig.PARTITION_FIELDS, tableConfig.getPartitionFieldProp());
     }
   }
+
+  static void downgradeMetadataPartitions(HoodieEngineContext context,
+                                          HoodieStorage hoodieStorage,
+                                          HoodieTableMetaClient mdtMetaClient,
+                                          Map<ConfigProperty, String> tablePropsToAdd) {
+    // Fetch metadata partition paths.
+    List<String> metadataPartitions = FSUtils.getAllPartitionPaths(
+        context,
+        hoodieStorage,
+        mdtMetaClient.getBasePath(),
+        true);
+
+    // Delete partitions.
+    List<String> validPartitionPaths = deleteMetadataPartition(context, mdtMetaClient, metadataPartitions);
+
+    // Clean the configuration.
+    tablePropsToAdd.put(TABLE_METADATA_PARTITIONS, String.join(",", validPartitionPaths));
+  }
+
+  static List<String> deleteMetadataPartition(HoodieEngineContext context,
```

Comment:
We have a util - `HoodieTableMetadataUtil.deleteMetadataPartition`. Is it possible to reuse that?

Reply:
Yeah, this function is a wrapper of that one. We need to do filtering before that.